### DISCUSSION TITLE
Allow multiple files upload in the file fieldtype that mimic what Assets can do

### DISCUSSION THREAD

**Discussion by chunkitdesign** [ExpressionEngine/ExpressionEngine#3159] [Feature Requests]
Please allow multiple files upload in the file fieldtype like what Assets offers. It would be very useful for creating a gallery like what you see in the screenshot.

<img width="1062" alt="CleanShot 2023-03-24 at 12 42 25@2x" src="https://user-images.githubusercontent.com/3195497/227588203-2554caab-3c0d-478f-af36-8b65f451d47e.png">

**Comment by intoeetive** [comment]
The File fieldtype is designed to hold one file at a time.

If you need to build gallery-like content, you can use File Grid fieldtype (which also supports drag&drop upload of multiple files at once)

**Comment by chunkitdesign** [comment]
I understand that we can use File Grid. But imagine that I want to build a gallery with 100 images. then I need to add 100 rows and select or upload the file 100 times. However, this feature allows users to add images all at once. It would be very convenient. If EE wants the new File manager to replace the Assets, please consider to add this feature and allows other users to have discussion about this topic. thanks,

**Comment by chunkitdesign** [comment]
May you please reopen this topic?

**Comment by intoeetive** [comment]
So currently you can upload multiple images all-at-once into File Grid.

However you can not **select** multiple images in File Grid all-at-once. Which I think is what you need.

There is no chance we modify the existing behavior of File field, but we can certainly consider this functionality for File Grids.

I'll reopen the topic, but can you please edit the title and original description, stating what you need as you described in your latter comment

**Comment by chunkitdesign** [comment]
another reason I hope that EE can implement to allow multiple files into **a single field** rather than a File Grid field is that Bloqs does not allow any grid field type. Currently. with Assets, i can add Assets as a bloq. Since Assets allows multiple files (not a file grid), it is very useful and easy to create gallery like the screenshot that I posted. So technically what I requested is a totally different feature.
To sum it up, please allow multiple files upload in a **single** File fieldype rather than create multiple rows in a File Grid fieldtype. Imagine i have 200 images, based on the currency UI, there would be 200 thick rows on the page.
